File: 1248/1248.go
func countNumberOfNiceSubarrays(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := countNumberOfNiceSubarrays(root.Left)
	r := countNumberOfNiceSubarrays(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1248/1248.py
class Solution:
    def count_number_of_nice_subarrays(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.count_number_of_nice_subarrays(root.left), self.count_number_of_nice_subarrays(root.right))